test/OperationTest.cpp

```diff
@@ -134,3 +138,128 @@ TEST_F(OperationTestFixture, verifyCachePreventsInProgressState) {
           ParsedAsJson(HasKeyMatching("status", Eq("not started"))),
           ParsedAsJson(HasKeyMatching("status", Eq("fully materialized")))));
 }
+// _____________________________________________________________________________
+
+class StallForeverOperation : public Operation {
```

Comment:
Can you externalize those classes to an `OperationTestHelpers.h` header, I don't like these large class definitions in the middle of a file.